Allow management command to ignore missing file

marojenka opened the pull request:
QOL fix allowing rendervariations command to process missing files by logging error instead of complete break.

marojenka commented:
aaaand it contains a misprint and error that does not exist in python2. So prob not such a good idea.

marojenka commented:
python2 incompatibility bothers me but I don't see any other clean way to do it.
Probably IOSError sold work in fresh py2 and it actually make sense since different kind if problems might rise on working with files right? :thinking: 
So sure py2 isn't supported at this point but if the rest of the project is working under py2 FileNotFoundError can be replaced by IOSError.
Cheers.

codingjoe commented on stdimage/management/commands/rendervariations.py:
```diff
@@ -77,9 +89,16 @@ def render(field, images, count, replace, do_render):
+        if not igrnore_missing:
+            raise
+        else:
+            logger.error(str(e))
```
Please don't use the logger here. The command should write to `stderr`.
If you would want to log an exception, you should actually log exception and always provide your own message. The original exception and stack trace will be included anyways. You're log should provide additional information to why the exception was not raised.

codingjoe replied:
Anyhow, in this case please write to `self.stderr`.

marojenka replied:
I kinda was too used to trowing everything to the logger without additional thought. Yeah, I will change that *a bit later*.

codingjoe commented on stdimage/management/commands/rendervariations.py:
```diff
@@ -8,6 +9,8 @@
 
 from stdimage.utils import render_variations
 
+logger = logging.getLogger()
```
I explain further down why I don't think using a logger is a good idea. But as a side note here:
I would recommend to use the package name. That way people have only a single logger per package, that they can configure in their projects.

codingjoe replied:
I also don't know the default, I think you are getting the root logger here, where I think you wanted `__name__`. Anyhow, check my comment further down about `stderr`.

marojenka replied:
yeah u are right, I just missed that.
*quality code*

codingjoe commented on stdimage/management/commands/rendervariations.py:
```diff
@@ -24,8 +27,16 @@ def add_arguments(self, parser):
                             default=False,
                             help='Replace existing files.')
 
+        parser.add_argument('--igrnore-missing',
```
I would recommend to add a short version too, `-i`.

codingjoe commented on stdimage/management/commands/rendervariations.py:
```diff
@@ -24,8 +27,16 @@ def add_arguments(self, parser):
+                            action='store_true',
+                            dest='igrnore_missing',
+                            default=False,
+                            help='Do not rise exception on missing file '
```
`s/rise/raise/`

codingjoe replied:
How about something that does not include a negation:
```Ignore and skip missing file errors.```

Up in the help you might want to document, that the command will exit with non-zero exit code if a file is not found.

codingjoe commented on stdimage/management/commands/rendervariations.py:
```diff
@@ -77,9 +89,16 @@ def render(field, images, count, replace, do_render):
+            render_variations(**kwargs)
+    except FileNotFoundError as e:
+        if not igrnore_missing:
+            raise
```
I would recommend to cast to a `CommandError` and provide a more descriptive error message, eg:
```python
raise CommandError("Original file was not found, terminating command. Use --ignore-missing to skip this error.") from e
```

marojenka commented:
>  My bigger request would be to add a test 
I did some test https://github.com/codingjoe/django-stdimage/pull/191/commits/024a3401d6d769f2d8bf0d3a21bab5b4997fc5da#diff-59262c12118a93348565a41ab7f6ce42R86
not enough?

codingjoe commented:
> I did some test 024a340#diff-59262c12118a93348565a41ab7f6ce42R86
not enough?

That doesn't look bad, but the commit is not in your PR.

> the rest are reasonable comments, thanks.

thx, I try my best :)

marojenka commented:
it was some time but here is my changes to your comments.
I added shorted option -i, added it to the test (really important), ditched logger, raised CommandError.
M, forgot documentation.

codingjoe commented:
Released in 3.2.0
